[32mINFO    [0m juju.model:model.py:2098 Deploying ch:amd64/jammy/self-signed-certificates-155
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [idle] active: Primary
  shard-one/1 [idle] active: 
  shard-one/2 [idle] active: 
  shard-two/0 [idle] active: Primary
  config-server/0 [idle] active: 
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [executing] active: 
  shard-one/1 [executing] active: 
  shard-one/2 [executing] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/1 [executing] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [idle] active: 
  shard-one/1 [executing] active: 
  shard-one/2 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [idle] blocked: Shard requires TLS to be enabled.
  shard-one/1 [executing] blocked: Shard requires TLS to be enabled.
  shard-one/2 [executing] blocked: Shard requires TLS to be enabled.
  shard-two/0 [idle] active: Primary
  config-server/0 [idle] blocked: shards shard-one are unreachable.
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [idle] blocked: Shard requires TLS to be enabled.
  shard-one/1 [executing] blocked: Shard requires TLS to be enabled.
  shard-one/2 [executing] blocked: Shard requires TLS to be enabled.
  config-server/0 [idle] blocked: shards shard-one are unreachable.
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [idle] blocked: Shard requires TLS to be enabled.
  shard-one/1 [executing] blocked: Shard requires TLS to be enabled.
  shard-one/2 [executing] blocked: Shard requires TLS to be enabled.
  config-server/0 [idle] blocked: shards shard-one are unreachable.
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [executing] blocked: Shard requires TLS to be enabled.
  shard-one/1 [executing] blocked: Shard requires TLS to be enabled.
  shard-one/2 [executing] blocked: Shard requires TLS to be enabled.
  config-server/0 [idle] blocked: shards shard-one are unreachable.
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [executing] blocked: Shard requires TLS to be enabled.
  shard-one/1 [executing] blocked: Shard requires TLS to be enabled.
  shard-one/2 [executing] blocked: Shard requires TLS to be enabled.
  config-server/0 [idle] blocked: shards shard-one are unreachable.
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [executing] blocked: Shard requires TLS to be enabled.
  shard-one/1 [executing] blocked: Shard requires TLS to be enabled.
  shard-one/2 [executing] blocked: Shard requires TLS to be enabled.
  config-server/0 [idle] blocked: shards shard-one are unreachable.
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [executing] blocked: Shard requires TLS to be enabled.
  shard-one/1 [executing] blocked: Shard requires TLS to be enabled.
  shard-one/2 [executing] blocked: Shard requires TLS to be enabled.
  config-server/0 [idle] blocked: shards shard-one are unreachable.
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [executing] blocked: Shard requires TLS to be enabled.
  shard-one/1 [executing] blocked: Shard requires TLS to be enabled.
  shard-one/2 [executing] blocked: Shard requires TLS to be enabled.
  config-server/0 [idle] blocked: shards shard-one are unreachable.
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [executing] blocked: Shard requires TLS to be enabled.
  shard-one/1 [executing] blocked: Shard requires TLS to be enabled.
  shard-one/2 [executing] blocked: Shard requires TLS to be enabled.
  config-server/0 [idle] blocked: shards shard-one are unreachable.
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [executing] blocked: Shard requires TLS to be enabled.
  shard-one/1 [executing] blocked: Shard requires TLS to be enabled.
  shard-one/2 [executing] blocked: Shard requires TLS to be enabled.
  config-server/0 [idle] blocked: shards shard-one are unreachable.
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [executing] blocked: Shard requires TLS to be enabled.
  shard-one/1 [executing] blocked: Shard requires TLS to be enabled.
  shard-one/2 [executing] blocked: Shard requires TLS to be enabled.
  config-server/0 [idle] blocked: shards shard-one are unreachable.
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [executing] blocked: Shard requires TLS to be enabled.
  shard-one/1 [executing] blocked: Shard requires TLS to be enabled.
  shard-one/2 [executing] blocked: Shard requires TLS to be enabled.
  config-server/0 [idle] blocked: shards shard-one are unreachable.
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [executing] blocked: Shard requires TLS to be enabled.
  shard-one/1 [executing] blocked: Shard requires TLS to be enabled.
  shard-one/2 [executing] blocked: Shard requires TLS to be enabled.
  config-server/0 [idle] blocked: shards shard-one are unreachable.
  config-server/1 [idle] blocked: shards shard-one are unreachable.
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [executing] blocked: Shard requires TLS to be enabled.
  shard-one/1 [executing] blocked: Shard requires TLS to be enabled.
  shard-one/2 [executing] blocked: Shard requires TLS to be enabled.
  config-server/0 [idle] blocked: shards shard-one are unreachable.
  config-server/1 [idle] blocked: shards shard-one are unreachable.
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [executing] blocked: Shard requires TLS to be enabled.
  shard-one/1 [executing] blocked: Shard requires TLS to be enabled.
  shard-one/2 [executing] blocked: Shard requires TLS to be enabled.
  config-server/0 [idle] blocked: shards shard-one are unreachable.
  config-server/1 [idle] blocked: shards shard-one are unreachable.
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [executing] maintenance: enabling TLS
  shard-one/1 [executing] blocked: Shard requires TLS to be enabled.
  shard-one/2 [executing] blocked: Shard requires TLS to be enabled.
  config-server/0 [idle] blocked: shards shard-one are unreachable.
  config-server/1 [idle] blocked: shards shard-one are unreachable.
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [idle] active: 
  shard-one/1 [executing] blocked: Shard requires TLS to be enabled.
  shard-one/2 [executing] blocked: Shard requires TLS to be enabled.
  config-server/0 [idle] blocked: shards shard-one are unreachable.
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [idle] active: 
  shard-one/1 [executing] maintenance: enabling TLS
  shard-one/2 [executing] active: 
  config-server/0 [idle] blocked: shards shard-one are unreachable.
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/2 [idle] active: 
  config-server/0 [idle] blocked: shards shard-one are unreachable.
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  config-server/0 [idle] active: 
  config-server/1 [idle] active: Primary
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [idle] active: 
  shard-one/1 [idle] active: 
  shard-one/2 [idle] active: 
  shard-two/0 [idle] active: Primary
  config-server/0 [executing] active: 
  config-server/1 [executing] active: Primary
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [idle] active: 
  shard-one/1 [idle] active: 
  shard-one/2 [idle] active: 
  shard-two/0 [executing] active: Primary
  config-server/0 [idle] active: 
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [idle] blocked: Shard has TLS enabled, but config-server does not.
  shard-one/1 [idle] blocked: Shard has TLS enabled, but config-server does not.
  shard-one/2 [idle] blocked: Shard has TLS enabled, but config-server does not.
  shard-two/0 [idle] blocked: Shard has TLS enabled, but config-server does not.
  config-server/0 [executing] active: Primary
  config-server/1 [executing] active:
[32mINFO    [0m pytest_operator.plugin:plugin.py:903 Model status:

Model  Controller          Cloud/Region        Version  SLA          Timestamp
test   microk8s-localhost  microk8s/localhost  3.5.3    unsupported  02:29:56Z

App                                Version  Status   Scale  Charm                     Channel        Rev  Address         Exposed  Message
config-server                               error        2  mongodb-k8s                                3  10.152.183.101  no       hook failed: "secret-changed"
self-signed-certificates                    active       1  self-signed-certificates  latest/stable  155  10.152.183.175  no       
self-signed-certificates-separate           active       1  self-signed-certificates  latest/stable  155  10.152.183.76   no       
shard-one                                   blocked      3  mongodb-k8s                                4  10.152.183.118  no       Shard has TLS enabled, but config-server does not.
shard-two                                   blocked      1  mongodb-k8s                                5  10.152.183.119  no       Shard has TLS enabled, but config-server does not.

Unit                                  Workload     Agent      Address      Ports  Message
config-server/0                       error        idle       10.1.13.97          hook failed: "secret-changed"
config-server/1*                      maintenance  executing  10.1.13.98          enabling TLS
self-signed-certificates-separate/0*  active       idle       10.1.13.111         
self-signed-certificates/0*           active       idle       10.1.13.90          
shard-one/0*                          blocked      executing  10.1.13.104         Shard has TLS enabled, but config-server does not.
shard-one/1                           blocked      executing  10.1.13.107         Shard has TLS enabled, but config-server does not.
shard-one/2                           blocked      executing  10.1.13.106         Shard has TLS enabled, but config-server does not.
shard-two/0*                          blocked      executing  10.1.13.110         Shard has TLS enabled, but config-server does not.

[32mINFO    [0m pytest_operator.plugin:plugin.py:909 Juju error logs:

unit-shard-one-0: 01:37:50 ERROR unit.shard-one/0.juju-log database-peers:1: Charm is in sharding role: database. Does not support shard interface.
unit-shard-one-0: 01:37:56 ERROR unit.shard-one/0.juju-log database-peers:1: Charm is in sharding role: database. Does not support shard interface.
unit-config-server-0: 01:39:15 ERROR unit.config-server/0.juju-log config-server:3: Failed to add shard shard-one to the config server, error=OperationFailure('Authentication failed., full error: {\'ok\': 0.0, \'errmsg\': \'Authentication failed.\', \'code\': 18, \'codeName\': \'AuthenticationFailed\', \'$clusterTime\': {\'clusterTime\': Timestamp(1727314755, 2), \'signature\': {\'hash\': b"M2\\xab]\\x1a:p7\'x\\x87C>\\xf3\\x9cq\\x0e\\xe2\\x85\\xd5", \'keyId\': 7418759905881882649}}, \'operationTime\': Timestamp(1727314755, 2)}')
unit-config-server-0: 01:39:16 ERROR unit.config-server/0.juju-log config-server:3: shard-one shard does not have the same auth as the config server.
unit-config-server-0: 01:39:32 ERROR unit.config-server/0.juju-log config-server:4: Failed to add shard shard-one to the config server, error=OperationFailure('Could not find host matching read preference { mode: "primary" } for set shard-one, full error: {\'ok\': 0.0, \'errmsg\': \'Could not find host matching read preference { mode: "primary" } for set shard-one\', \'code\': 133, \'codeName\': \'FailedToSatisfyReadPreference\', \'$clusterTime\': {\'clusterTime\': Timestamp(1727314771, 1), \'signature\': {\'hash\': b\'@:QS\\xca\\xe8C\\xd22\\xd9\\xff\\x10\\xe4OI\\x1a\\x15\\xe9\\xe9\\xf8\', \'keyId\': 7418759905881882649}}, \'operationTime\': Timestamp(1727314771, 1)}')
unit-config-server-0: 01:39:32 ERROR unit.config-server/0.juju-log config-server:4: Failed to add shard-one to cluster
unit-config-server-0: 01:39:32 ERROR unit.config-server/0.juju-log config-server:4: Deferring _on_relation_event for shards interface since: error=OperationFailure('Could not find host matching read preference { mode: "primary" } for set shard-one, full error: {\'ok\': 0.0, \'errmsg\': \'Could not find host matching read preference { mode: "primary" } for set shard-one\', \'code\': 133, \'codeName\': \'FailedToSatisfyReadPreference\', \'$clusterTime\': {\'clusterTime\': Timestamp(1727314771, 1), \'signature\': {\'hash\': b\'@:QS\\xca\\xe8C\\xd22\\xd9\\xff\\x10\\xe4OI\\x1a\\x15\\xe9\\xe9\\xf8\', \'keyId\': 7418759905881882649}}, \'operationTime\': Timestamp(1727314771, 1)}')
unit-config-server-0: 01:39:35 ERROR unit.config-server/0.juju-log config-server:4: Failed to add shard shard-two to the config server, error=OperationFailure("Authentication failed., full error: {'ok': 0.0, 'errmsg': 'Authentication failed.', 'code': 18, 'codeName': 'AuthenticationFailed', '$clusterTime': {'clusterTime': Timestamp(1727314774, 1), 'signature': {'hash': b'\\xcc\\xe5\\xd0\\x9b\\xd08P|\\xba\\\\\\xbb\\x9d\\xb3V\\xaa\\xb4:\\x99\\x13\\x07', 'keyId': 7418759905881882649}}, 'operationTime': Timestamp(1727314774, 1)}")
unit-config-server-0: 01:39:35 ERROR unit.config-server/0.juju-log config-server:4: shard-two shard does not have the same auth as the config server.
unit-config-server-0: 01:39:37 ERROR unit.config-server/0.juju-log config-server:4: Failed to add shard shard-two to the config server, error=OperationFailure("Authentication failed., full error: {'ok': 0.0, 'errmsg': 'Authentication failed.', 'code': 18, 'codeName': 'AuthenticationFailed', '$clusterTime': {'clusterTime': Timestamp(1727314776, 1), 'signature': {'hash': b'Em,\\xa6v\\xc7\\x9b\\x92\\xa9\\x92\\xf5\\xa8\\xef<^KT9k\\xe4', 'keyId': 7418759905881882649}}, 'operationTime': Timestamp(1727314776, 1)}")
unit-config-server-0: 01:39:37 ERROR unit.config-server/0.juju-log config-server:4: shard-two shard does not have the same auth as the config server.
unit-shard-one-1: 01:48:47 ERROR unit.shard-one/1.juju-log certificates:5: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-shard-one-2: 01:48:47 ERROR unit.shard-one/2.juju-log certificates:5: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-shard-one-1: 01:48:48 ERROR unit.shard-one/1.juju-log certificates:5: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-config-server-1: 01:48:48 ERROR unit.config-server/1.juju-log certificates:7: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-shard-one-2: 01:48:48 ERROR unit.shard-one/2.juju-log certificates:5: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-shard-two-0: 01:48:48 ERROR unit.shard-two/0.juju-log certificates:6: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-config-server-0: 01:48:48 ERROR unit.config-server/0.juju-log certificates:7: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-shard-one-0: 01:48:48 ERROR unit.shard-one/0.juju-log certificates:5: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-shard-two-0: 01:48:48 ERROR unit.shard-two/0.juju-log certificates:6: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-config-server-0: 01:48:49 ERROR unit.config-server/0.juju-log certificates:7: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-config-server-1: 01:48:49 ERROR unit.config-server/1.juju-log certificates:7: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-shard-one-0: 01:48:49 ERROR unit.shard-one/0.juju-log certificates:5: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-config-server-1: 01:49:00 ERROR unit.config-server/1.juju-log certificates:7: This operation (update_relation_data()) can only be performed by the leader unit
unit-config-server-1: 01:49:00 ERROR unit.config-server/1.juju-log certificates:7: This operation (update_relation_data()) can only be performed by the leader unit
unit-config-server-1: 01:49:15 ERROR unit.config-server/1.juju-log This operation (update_relation_data()) can only be performed by the leader unit
unit-config-server-1: 01:49:15 ERROR unit.config-server/1.juju-log This operation (update_relation_data()) can only be performed by the leader unit
unit-shard-one-0: 01:49:58 ERROR unit.shard-one/0.juju-log Unable to access primary due to: No replica set members match selector "Primary()", Timeout: 1.0s, Topology Description: <TopologyDescription id: 66f4bdc59255e10ec2f443e6, topology_type: ReplicaSetNoPrimary, servers: [<ServerDescription ('shard-one-0.shard-one-endpoints', 27017) server_type: RSSecondary, rtt: 0.0007481300001472846>, <ServerDescription ('shard-one-1.shard-one-endpoints', 27017) server_type: RSSecondary, rtt: 0.0005013169998164813>, <ServerDescription ('shard-one-2.shard-one-endpoints', 27017) server_type: RSSecondary, rtt: 0.0007834549999188312>]>
unit-shard-one-0: 01:50:01 ERROR unit.shard-one/0.juju-log Unable to access primary due to: No replica set members match selector "Primary()", Timeout: 1.0s, Topology Description: <TopologyDescription id: 66f4bdc7217b98ad7a9c1ce1, topology_type: ReplicaSetNoPrimary, servers: [<ServerDescription ('shard-one-0.shard-one-endpoints', 27017) server_type: RSSecondary, rtt: 0.0004625100000339444>, <ServerDescription ('shard-one-1.shard-one-endpoints', 27017) server_type: Unknown, rtt: None, error=NotPrimaryError('The server is in quiesce mode and will shut down, full error: {\'topologyVersion\': {\'processId\': ObjectId(\'66f4bdc2ea78ef37d2504ba6\'), \'counter\': 4}, \'ok\': 0.0, \'errmsg\': \'The server is in quiesce mode and will shut down\', \'code\': 91, \'codeName\': \'ShutdownInProgress\', \'remainingQuiesceTimeMillis\': 14493, \'lastCommittedOpTime\': Timestamp(0, 0), \'$clusterTime\': {\'clusterTime\': Timestamp(1727315396, 1), \'signature\': {\'hash\': b\'\\xafMY\\xbc\\x00\\x84C\\x8b\\xec\\x9c\\xc1\\xbe\\x89\\xdb\\x1cps\\xa7"C\', \'keyId\': 7418759905881882649}}, \'operationTime\': Timestamp(1727315332, 1)}')>, <ServerDescription ('shard-one-2.shard-one-endpoints', 27017) server_type: RSSecondary, rtt: 0.0007172210000589985>]>
unit-shard-one-0: 01:51:13 ERROR unit.shard-one/0.juju-log certificates:5: An unknown certificate is available -- ignoring.
unit-shard-one-0: 01:51:13 ERROR unit.shard-one/0.juju-log certificates:5: An unknown certificate is available -- ignoring.
unit-shard-one-1: 01:51:15 ERROR unit.shard-one/1.juju-log certificates:5: An unknown certificate is available -- ignoring.
unit-shard-one-1: 01:51:15 ERROR unit.shard-one/1.juju-log certificates:5: An unknown certificate is available -- ignoring.
unit-shard-one-2: 01:51:34 ERROR unit.shard-one/2.juju-log certificates:5: An unknown certificate is available -- ignoring.
unit-shard-one-2: 01:51:34 ERROR unit.shard-one/2.juju-log certificates:5: An unknown certificate is available -- ignoring.
unit-shard-one-0: 01:51:35 ERROR unit.shard-one/0.juju-log certificates:5: An unknown certificate is available -- ignoring.
unit-shard-one-0: 01:51:35 ERROR unit.shard-one/0.juju-log certificates:5: An unknown certificate is available -- ignoring.
unit-shard-one-1: 01:51:37 ERROR unit.shard-one/1.juju-log certificates:5: An unknown certificate is available -- ignoring.
unit-shard-one-1: 01:51:37 ERROR unit.shard-one/1.juju-log certificates:5: An unknown certificate is available -- ignoring.
unit-shard-two-0: 01:53:23 ERROR unit.shard-two/0.juju-log certificates:6: An unknown certificate is available -- ignoring.
unit-shard-two-0: 01:53:23 ERROR unit.shard-two/0.juju-log certificates:6: An unknown certificate is available -- ignoring.
unit-config-server-0: 01:55:03 ERROR unit.config-server/0.juju-log certificates:7: An unknown certificate is available -- ignoring.
unit-config-server-0: 01:55:03 ERROR unit.config-server/0.juju-log certificates:7: An unknown certificate is available -- ignoring.
unit-config-server-1: 01:55:05 ERROR unit.config-server/1.juju-log certificates:7: An unknown certificate is available -- ignoring.
unit-config-server-1: 01:55:05 ERROR unit.config-server/1.juju-log certificates:7: An unknown certificate is available -- ignoring.
unit-config-server-1: 01:55:06 ERROR unit.config-server/1.juju-log certificates:7: An unknown certificate is available -- ignoring.
unit-config-server-1: 01:55:06 ERROR unit.config-server/1.juju-log certificates:7: An unknown certificate is available -- ignoring.
unit-config-server-1: 01:55:07 ERROR unit.config-server/1.juju-log certificates:7: This operation (update_relation_data()) can only be performed by the leader unit
unit-config-server-1: 01:55:07 ERROR unit.config-server/1.juju-log certificates:7: This operation (update_relation_data()) can only be performed by the leader unit
unit-config-server-0: 01:55:23 ERROR unit.config-server/0.juju-log certificates:7: An unknown certificate is available -- ignoring.
unit-config-server-0: 01:55:23 ERROR unit.config-server/0.juju-log certificates:7: An unknown certificate is available -- ignoring.
unit-config-server-1: 01:57:11 ERROR unit.config-server/1.juju-log certificates:7: This operation (delete_relation_data()) can only be performed by the leader unit
unit-config-server-1: 01:57:11 ERROR unit.config-server/1.juju-log certificates:7: This operation (delete_relation_data()) can only be performed by the leader unit
unit-shard-two-0: 01:57:23 ERROR juju.worker.dependency "uniter" manifold worker returned unexpected error: relation scope id "980fc863-5d7c-484a-884a-e4ebedd6404b:r#6#self-signed-certificates": settings 980fc863-5d7c-484a-884a-e4ebedd6404b:r#6#self-signed-certificates not found (not found)
unit-shard-one-0: 01:57:27 ERROR unit.shard-one/0.juju-log Unable to access primary due to: shard-one-1.shard-one-endpoints:27017: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms),shard-one-2.shard-one-endpoints:27017: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms),shard-one-0.shard-one-endpoints:27017: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms), Timeout: 1.0s, Topology Description: <TopologyDescription id: 66f4bf863085e79db8f93e4f, topology_type: ReplicaSetNoPrimary, servers: [<ServerDescription ('shard-one-0.shard-one-endpoints', 27017) server_type: Unknown, rtt: None, error=AutoReconnect('shard-one-0.shard-one-endpoints:27017: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms)')>, <ServerDescription ('shard-one-1.shard-one-endpoints', 27017) server_type: Unknown, rtt: None, error=AutoReconnect('shard-one-1.shard-one-endpoints:27017: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms)')>, <ServerDescription ('shard-one-2.shard-one-endpoints', 27017) server_type: Unknown, rtt: None, error=AutoReconnect('shard-one-2.shard-one-endpoints:27017: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms)')>]>
unit-shard-one-0: 01:57:30 ERROR unit.shard-one/0.juju-log Unable to access primary due to: shard-one-1.shard-one-endpoints:27017: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms),shard-one-0.shard-one-endpoints:27017: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms),shard-one-2.shard-one-endpoints:27017: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms), Timeout: 1.0s, Topology Description: <TopologyDescription id: 66f4bf893ea27068fb61dd00, topology_type: ReplicaSetNoPrimary, servers: [<ServerDescription ('shard-one-0.shard-one-endpoints', 27017) server_type: Unknown, rtt: None, error=AutoReconnect('shard-one-0.shard-one-endpoints:27017: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms)')>, <ServerDescription ('shard-one-1.shard-one-endpoints', 27017) server_type: Unknown, rtt: None, error=AutoReconnect('shard-one-1.shard-one-endpoints:27017: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms)')>, <ServerDescription ('shard-one-2.shard-one-endpoints', 27017) server_type: Unknown, rtt: None, error=AutoReconnect('shard-one-2.shard-one-endpoints:27017: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms)')>]>
unit-config-server-0: 01:57:30 ERROR unit.config-server/0.juju-log config-server:3: Deferring _on_relation_event for shards interface since: error=ServerSelectionTimeoutError("config-server-0.config-server-endpoints:27018: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms),config-server-1.config-server-endpoints:27018: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms), Timeout: 1.0s, Topology Description: <TopologyDescription id: 66f4bf89b8abd252560b405e, topology_type: Unknown, servers: [<ServerDescription ('config-server-0.config-server-endpoints', 27018) server_type: Unknown, rtt: None, error=AutoReconnect('config-server-0.config-server-endpoints:27018: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms)')>, <ServerDescription ('config-server-1.config-server-endpoints', 27018) server_type: Unknown, rtt: None, error=AutoReconnect('config-server-1.config-server-endpoints:27018: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms)')>]>")
unit-shard-one-0: 01:57:58 ERROR unit.shard-one/0.juju-log database-peers:1: Charm is in sharding role: database. Does not support shard interface.
unit-shard-one-0: 01:58:01 ERROR unit.shard-one/0.juju-log database-peers:1: Charm is in sharding role: database. Does not support shard interface.
unit-shard-one-1: 01:58:01 ERROR unit.shard-one/1.juju-log shard-one/1.is_unit_in_replica_set PyMongoError=No replica set members match selector "Primary()", Timeout: 1.0s, Topology Description: <TopologyDescription id: 66f4bfa8940192e7dfe1f2bc, topology_type: ReplicaSetNoPrimary, servers: [<ServerDescription ('shard-one-1.shard-one-endpoints', 27017) server_type: RSGhost, rtt: 0.00148264599988579>]>
unit-config-server-1: 01:58:01 ERROR unit.config-server/1.juju-log config-server/1.is_unit_in_replica_set PyMongoError=No replica set members match selector "Primary()", Timeout: 1.0s, Topology Description: <TopologyDescription id: 66f4bfa8a7d2fcc16a9a52e5, topology_type: ReplicaSetNoPrimary, servers: [<ServerDescription ('config-server-1.config-server-endpoints', 27017) server_type: RSGhost, rtt: 0.003068751999990127>]>
controller-0: 01:59:12 ERROR juju.worker.caasapplicationprovisioner.runner exited "config-server": finding filesystem info for config-server-mongodb-logs-3d4baab2: unable to get persistent volume claim: k8s: persistentvolumeclaims "config-server-mongodb-logs-3d4baab2-config-server-0" not found
unit-shard-one-0: 02:00:52 ERROR unit.shard-one/0.juju-log database-peers:9: Charm is in sharding role: database. Does not support shard interface.
unit-shard-one-0: 02:00:58 ERROR unit.shard-one/0.juju-log database-peers:9: Charm is in sharding role: database. Does not support shard interface.
unit-config-server-1: 02:01:29 ERROR unit.config-server/1.juju-log certificates:13: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-shard-one-0: 02:01:29 ERROR unit.shard-one/0.juju-log certificates:11: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-shard-one-2: 02:01:29 ERROR unit.shard-one/2.juju-log certificates:11: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-config-server-0: 02:01:29 ERROR unit.config-server/0.juju-log certificates:13: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-shard-one-1: 02:01:29 ERROR unit.shard-one/1.juju-log certificates:11: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-shard-two-0: 02:01:29 ERROR unit.shard-two/0.juju-log certificates:12: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-shard-one-0: 02:01:30 ERROR unit.shard-one/0.juju-log certificates:11: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-shard-one-2: 02:01:30 ERROR unit.shard-one/2.juju-log certificates:11: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-config-server-0: 02:01:30 ERROR unit.config-server/0.juju-log certificates:13: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-shard-one-1: 02:01:30 ERROR unit.shard-one/1.juju-log certificates:11: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-config-server-1: 02:01:30 ERROR unit.config-server/1.juju-log certificates:13: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-shard-two-0: 02:01:30 ERROR unit.shard-two/0.juju-log certificates:12: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-config-server-1: 02:02:52 ERROR unit.config-server/1.juju-log config-server:14: Failed to add shard shard-one to the config server, error=OperationFailure('Could not find user "O=config-server,x500UniqueIdentifier=5aabaf0c-9668-44db-bc73-cf984b41f838,CN=config-server" for db "$external", full error: {\'ok\': 0.0, \'errmsg\': \'Could not find user "O=config-server,x500UniqueIdentifier=5aabaf0c-9668-44db-bc73-cf984b41f838,CN=config-server" for db "$external"\', \'code\': 11, \'codeName\': \'UserNotFound\', \'$clusterTime\': {\'clusterTime\': Timestamp(1727316170, 3), \'signature\': {\'hash\': b\'B1\\x15\\x00Q1\\xaa{\\xd9DV\\xfeUN1\\x1f\\xed\\x03u\\xef\', \'keyId\': 7418765811461914649}}, \'operationTime\': Timestamp(1727316170, 3)}')
unit-config-server-1: 02:02:52 ERROR unit.config-server/1.juju-log config-server:14: Failed to add shard-one to cluster
unit-config-server-1: 02:02:52 ERROR unit.config-server/1.juju-log config-server:14: Deferring _on_relation_event for shards interface since: error=OperationFailure('Could not find user "O=config-server,x500UniqueIdentifier=5aabaf0c-9668-44db-bc73-cf984b41f838,CN=config-server" for db "$external", full error: {\'ok\': 0.0, \'errmsg\': \'Could not find user "O=config-server,x500UniqueIdentifier=5aabaf0c-9668-44db-bc73-cf984b41f838,CN=config-server" for db "$external"\', \'code\': 11, \'codeName\': \'UserNotFound\', \'$clusterTime\': {\'clusterTime\': Timestamp(1727316170, 3), \'signature\': {\'hash\': b\'B1\\x15\\x00Q1\\xaa{\\xd9DV\\xfeUN1\\x1f\\xed\\x03u\\xef\', \'keyId\': 7418765811461914649}}, \'operationTime\': Timestamp(1727316170, 3)}')
unit-config-server-1: 02:02:53 ERROR unit.config-server/1.juju-log config-server:14: Failed to add shard shard-one to the config server, error=OperationFailure('Could not find user "O=config-server,x500UniqueIdentifier=5aabaf0c-9668-44db-bc73-cf984b41f838,CN=config-server" for db "$external", full error: {\'ok\': 0.0, \'errmsg\': \'Could not find user "O=config-server,x500UniqueIdentifier=5aabaf0c-9668-44db-bc73-cf984b41f838,CN=config-server" for db "$external"\', \'code\': 11, \'codeName\': \'UserNotFound\', \'$clusterTime\': {\'clusterTime\': Timestamp(1727316173, 1), \'signature\': {\'hash\': b\'\\x82$A\\xe1\\xc8K\\x15\\xecm\\x9bM\\xec\\xbbj\\xa5izT\\xec\\x02\', \'keyId\': 7418765811461914649}}, \'operationTime\': Timestamp(1727316173, 1)}')
unit-config-server-1: 02:02:53 ERROR unit.config-server/1.juju-log config-server:14: Failed to add shard-one to cluster
unit-config-server-1: 02:02:53 ERROR unit.config-server/1.juju-log config-server:14: Deferring _on_relation_event for shards interface since: error=OperationFailure('Could not find user "O=config-server,x500UniqueIdentifier=5aabaf0c-9668-44db-bc73-cf984b41f838,CN=config-server" for db "$external", full error: {\'ok\': 0.0, \'errmsg\': \'Could not find user "O=config-server,x500UniqueIdentifier=5aabaf0c-9668-44db-bc73-cf984b41f838,CN=config-server" for db "$external"\', \'code\': 11, \'codeName\': \'UserNotFound\', \'$clusterTime\': {\'clusterTime\': Timestamp(1727316173, 1), \'signature\': {\'hash\': b\'\\x82$A\\xe1\\xc8K\\x15\\xecm\\x9bM\\xec\\xbbj\\xa5izT\\xec\\x02\', \'keyId\': 7418765811461914649}}, \'operationTime\': Timestamp(1727316173, 1)}')
unit-shard-one-1: 02:03:07 ERROR unit.shard-one/1.juju-log certificates:11: An unknown certificate is available -- ignoring.
unit-shard-one-1: 02:03:07 ERROR unit.shard-one/1.juju-log certificates:11: An unknown certificate is available -- ignoring.
unit-config-server-1: 02:03:10 ERROR unit.config-server/1.juju-log config-server:14: Failed to add shard shard-one to the config server, error=OperationFailure('Could not find host matching read preference { mode: "primary" } for set shard-one, full error: {\'ok\': 0.0, \'errmsg\': \'Could not find host matching read preference { mode: "primary" } for set shard-one\', \'code\': 133, \'codeName\': \'FailedToSatisfyReadPreference\', \'$clusterTime\': {\'clusterTime\': Timestamp(1727316189, 1), \'signature\': {\'hash\': b\'x\\xf5@<_\\x9bY\\xe1\\x0e\\x89)\\xd0b\\xb1u\\x8f1H\\t\\x83\', \'keyId\': 7418765811461914649}}, \'operationTime\': Timestamp(1727316189, 1)}')
unit-config-server-1: 02:03:10 ERROR unit.config-server/1.juju-log config-server:14: Failed to add shard-one to cluster
unit-config-server-1: 02:03:10 ERROR unit.config-server/1.juju-log config-server:14: Deferring _on_relation_event for shards interface since: error=OperationFailure('Could not find host matching read preference { mode: "primary" } for set shard-one, full error: {\'ok\': 0.0, \'errmsg\': \'Could not find host matching read preference { mode: "primary" } for set shard-one\', \'code\': 133, \'codeName\': \'FailedToSatisfyReadPreference\', \'$clusterTime\': {\'clusterTime\': Timestamp(1727316189, 1), \'signature\': {\'hash\': b\'x\\xf5@<_\\x9bY\\xe1\\x0e\\x89)\\xd0b\\xb1u\\x8f1H\\t\\x83\', \'keyId\': 7418765811461914649}}, \'operationTime\': Timestamp(1727316189, 1)}')
unit-shard-one-2: 02:03:11 ERROR unit.shard-one/2.juju-log certificates:11: An unknown certificate is available -- ignoring.
unit-shard-one-2: 02:03:11 ERROR unit.shard-one/2.juju-log certificates:11: An unknown certificate is available -- ignoring.
unit-shard-one-0: 02:03:12 ERROR unit.shard-one/0.juju-log certificates:11: An unknown certificate is available -- ignoring.
unit-shard-one-0: 02:03:12 ERROR unit.shard-one/0.juju-log certificates:11: An unknown certificate is available -- ignoring.
unit-config-server-1: 02:03:25 ERROR unit.config-server/1.juju-log config-server:14: Failed to add shard shard-one to the config server, error=OperationFailure('Could not find host matching read preference { mode: "primary" } for set shard-one, full error: {\'ok\': 0.0, \'errmsg\': \'Could not find host matching read preference { mode: "primary" } for set shard-one\', \'code\': 133, \'codeName\': \'FailedToSatisfyReadPreference\', \'$clusterTime\': {\'clusterTime\': Timestamp(1727316204, 1), \'signature\': {\'hash\': b\'yh\\x06\\xf7\\xeb\\xe9nP\\xf8v"\\x19\\x03{\\xf3\\x8f\\xf0\\xbcv\\xf2\', \'keyId\': 7418765811461914649}}, \'operationTime\': Timestamp(1727316204, 1)}')
unit-config-server-1: 02:03:25 ERROR unit.config-server/1.juju-log config-server:14: Failed to add shard-one to cluster
unit-config-server-1: 02:03:25 ERROR unit.config-server/1.juju-log config-server:14: Deferring _on_relation_event for shards interface since: error=OperationFailure('Could not find host matching read preference { mode: "primary" } for set shard-one, full error: {\'ok\': 0.0, \'errmsg\': \'Could not find host matching read preference { mode: "primary" } for set shard-one\', \'code\': 133, \'codeName\': \'FailedToSatisfyReadPreference\', \'$clusterTime\': {\'clusterTime\': Timestamp(1727316204, 1), \'signature\': {\'hash\': b\'yh\\x06\\xf7\\xeb\\xe9nP\\xf8v"\\x19\\x03{\\xf3\\x8f\\xf0\\xbcv\\xf2\', \'keyId\': 7418765811461914649}}, \'operationTime\': Timestamp(1727316204, 1)}')
unit-config-server-1: 02:03:38 ERROR unit.config-server/1.juju-log config-server:15: Failed to add shard shard-two to the config server, error=OperationFailure('Could not find user "O=config-server,x500UniqueIdentifier=5aabaf0c-9668-44db-bc73-cf984b41f838,CN=config-server" for db "$external", full error: {\'ok\': 0.0, \'errmsg\': \'Could not find user "O=config-server,x500UniqueIdentifier=5aabaf0c-9668-44db-bc73-cf984b41f838,CN=config-server" for db "$external"\', \'code\': 11, \'codeName\': \'UserNotFound\', \'$clusterTime\': {\'clusterTime\': Timestamp(1727316217, 3), \'signature\': {\'hash\': b\'\\xa1\\xfd\\x0c\\xb8\\x8f2\\xf8\\xa1>\\xe4U,\\x9e\\x97\\xa6i<\\x00\\n\\xe4\', \'keyId\': 7418765811461914649}}, \'operationTime\': Timestamp(1727316217, 3)}')
unit-config-server-1: 02:03:39 ERROR unit.config-server/1.juju-log config-server:15: Failed to add shard-two to cluster
unit-config-server-1: 02:03:39 ERROR unit.config-server/1.juju-log config-server:15: Deferring _on_relation_event for shards interface since: error=OperationFailure('Could not find user "O=config-server,x500UniqueIdentifier=5aabaf0c-9668-44db-bc73-cf984b41f838,CN=config-server" for db "$external", full error: {\'ok\': 0.0, \'errmsg\': \'Could not find user "O=config-server,x500UniqueIdentifier=5aabaf0c-9668-44db-bc73-cf984b41f838,CN=config-server" for db "$external"\', \'code\': 11, \'codeName\': \'UserNotFound\', \'$clusterTime\': {\'clusterTime\': Timestamp(1727316217, 3), \'signature\': {\'hash\': b\'\\xa1\\xfd\\x0c\\xb8\\x8f2\\xf8\\xa1>\\xe4U,\\x9e\\x97\\xa6i<\\x00\\n\\xe4\', \'keyId\': 7418765811461914649}}, \'operationTime\': Timestamp(1727316217, 3)}')
unit-shard-two-0: 02:03:50 ERROR unit.shard-two/0.juju-log certificates:12: An unknown certificate is available -- ignoring.
unit-shard-two-0: 02:03:50 ERROR unit.shard-two/0.juju-log certificates:12: An unknown certificate is available -- ignoring.
unit-shard-one-1: 02:05:37 ERROR unit.shard-one/1.juju-log certificates:11: An unknown certificate is available -- ignoring.
unit-shard-one-1: 02:05:37 ERROR unit.shard-one/1.juju-log certificates:11: An unknown certificate is available -- ignoring.
unit-self-signed-certificates-0: 02:12:08 ERROR juju.worker.dependency "uniter" manifold worker returned unexpected error: relation scope id "980fc863-5d7c-484a-884a-e4ebedd6404b:r#11#shard-one": settings 980fc863-5d7c-484a-884a-e4ebedd6404b:r#11#shard-one not found (not found)
unit-shard-one-0: 02:12:10 ERROR juju.worker.dependency "uniter" manifold worker returned unexpected error: relation scope id "980fc863-5d7c-484a-884a-e4ebedd6404b:r#11#self-signed-certificates": settings 980fc863-5d7c-484a-884a-e4ebedd6404b:r#11#self-signed-certificates not found (not found)
unit-shard-one-1: 02:12:10 ERROR juju.worker.dependency "uniter" manifold worker returned unexpected error: relation scope id "980fc863-5d7c-484a-884a-e4ebedd6404b:r#11#self-signed-certificates": settings 980fc863-5d7c-484a-884a-e4ebedd6404b:r#11#self-signed-certificates not found (not found)
unit-shard-one-2: 02:12:10 ERROR juju.worker.dependency "uniter" manifold worker returned unexpected error: relation scope id "980fc863-5d7c-484a-884a-e4ebedd6404b:r#11#self-signed-certificates": settings 980fc863-5d7c-484a-884a-e4ebedd6404b:r#11#self-signed-certificates not found (not found)
unit-shard-one-0: 02:20:57 ERROR unit.shard-one/0.juju-log certificates:16: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-shard-one-0: 02:20:57 ERROR unit.shard-one/0.juju-log certificates:16: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-shard-one-1: 02:21:36 ERROR unit.shard-one/1.juju-log certificates:16: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-shard-one-1: 02:21:36 ERROR unit.shard-one/1.juju-log certificates:16: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-shard-one-2: 02:21:38 ERROR unit.shard-one/2.juju-log certificates:16: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-shard-one-2: 02:21:38 ERROR unit.shard-one/2.juju-log certificates:16: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-config-server-0: 02:27:26 ERROR unit.config-server/0.juju-log certificates:13: This operation (delete_relation_data()) can only be performed by the leader unit
unit-config-server-0: 02:27:26 ERROR unit.config-server/0.juju-log certificates:13: This operation (delete_relation_data()) can only be performed by the leader unit
unit-config-server-0: 02:29:32 ERROR unit.config-server/0.juju-log certificates:17: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-config-server-0: 02:29:32 ERROR unit.config-server/0.juju-log certificates:17: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-config-server-1: 02:29:33 ERROR unit.config-server/1.juju-log certificates:17: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-config-server-1: 02:29:34 ERROR unit.config-server/1.juju-log certificates:17: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-config-server-0: 02:29:34 ERROR unit.config-server/0.juju-log certificates:17: This operation (update_relation_data()) can only be performed by the leader unit
unit-config-server-0: 02:29:34 ERROR unit.config-server/0.juju-log certificates:17: This operation (update_relation_data()) can only be performed by the leader unit
unit-config-server-0: 02:29:49 ERROR unit.config-server/0.juju-log This operation (update_relation_data()) can only be performed by the leader unit
unit-config-server-0: 02:29:49 ERROR unit.config-server/0.juju-log This operation (update_relation_data()) can only be performed by the leader unit
unit-config-server-0: 02:29:55 ERROR unit.config-server/0.juju-log Uncaught exception while in charm code:
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-config-server-0/charm/./src/charm.py", line 1643, in <module>
    main(MongoDBCharm)
  File "/var/lib/juju/agents/unit-config-server-0/charm/venv/ops/main.py", line 553, in main
    manager.run()
  File "/var/lib/juju/agents/unit-config-server-0/charm/venv/ops/main.py", line 529, in run
    self._emit()
  File "/var/lib/juju/agents/unit-config-server-0/charm/venv/ops/main.py", line 515, in _emit
    self.framework.reemit()
  File "/var/lib/juju/agents/unit-config-server-0/charm/venv/ops/framework.py", line 863, in reemit
    self._reemit()
  File "/var/lib/juju/agents/unit-config-server-0/charm/venv/ops/framework.py", line 943, in _reemit
    custom_handler(event)
  File "/var/lib/juju/agents/unit-config-server-0/charm/lib/charms/mongodb/v1/mongodb_tls.py", line 237, in _on_certificate_available
    self.charm.restart_charm_services()
  File "/var/lib/juju/agents/unit-config-server-0/charm/./src/charm.py", line 1260, in restart_charm_services
    container.replan()
  File "/var/lib/juju/agents/unit-config-server-0/charm/venv/ops/model.py", line 2278, in replan
    self._pebble.replan_services()
  File "/var/lib/juju/agents/unit-config-server-0/charm/venv/ops/pebble.py", line 2131, in replan_services
    return self._services_action('replan', [], timeout, delay)
  File "/var/lib/juju/agents/unit-config-server-0/charm/venv/ops/pebble.py", line 2228, in _services_action
    raise ChangeError(change.err, change)
ops.pebble.ChangeError: cannot perform the following tasks:
- Start service "logrotate" (cannot start service: exited quickly with code 0)
----- Logs from task 0 -----
2024-09-26T02:29:55Z INFO Most recent service output:
    
2024-09-26T02:29:55Z ERROR cannot start service: exited quickly with code 0
-----
unit-config-server-0: 02:29:56 ERROR juju.worker.uniter.operation hook "secret-changed" (via hook dispatching script: dispatch) failed: exit status 1

[32mINFO    [0m pytest_operator.plugin:plugin.py:991 Forgetting model main...